third_party/txt/tests/paragraph_unittests.cc

```diff
@@ -1541,4 +1541,34 @@ TEST_F(RenderTest, RepeatLayoutParagraph) {
   ASSERT_TRUE(Snapshot());
 }
 
+TEST_F(RenderTest, Ellipsize) {
+  const char* text =
+      "This is a very long sentence to test if the text will properly wrap "
+      "around and go to the next line. Sometimes, short sentence. Longer "
+      "sentences are okay too because they are nessecary. Very short. ";
+  auto icu_text = icu::UnicodeString::fromUTF8(text);
+  std::u16string u16_text(icu_text.getBuffer(),
+                          icu_text.getBuffer() + icu_text.length());
+
+  txt::ParagraphStyle paragraph_style;
+  auto font_collection = FontCollection::GetFontCollection(txt::GetFontDir());
+  paragraph_style.ellipsis = u"\u2026";
+  txt::ParagraphBuilder builder(paragraph_style, &font_collection);
+
+  txt::TextStyle text_style;
+  text_style.color = SK_ColorBLACK;
+  builder.PushStyle(text_style);
+  builder.AddText(u16_text);
+
+  builder.Pop();
+
+  auto paragraph = builder.Build();
+  paragraph->Layout(GetTestCanvasWidth());
+
+  paragraph->Paint(GetCanvas(), 0, 0);
+
+  ASSERT_TRUE(Snapshot());
+  ASSERT_EQ(paragraph->records_.size(), 1ull);
```

Comment:
The test doesn't seem to be actually asserting the fact that an ellipsis was added (other than inspecting the snapshot). Can we look inside the record somehow to verify?

Reply:
The test checks that the layout was limited to one line instead of being wrapped onto a second line.  Added a comment about this.

I don't know of a way to check for the ellipsis - the Skia text blob doesn't expose the individual glyphs.